File: 27-RemoveElement/27-RemoveElement.cpp
// Last updated: 1/13/2026, 10:49:57 PM
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int i = 0;
        while (i < n) {
            if (nums[i] == val) {

                for (int j = i + 1; j < n; ++j) {
                    nums[j - 1] = nums[j];
                }
                --n;
            } else {
                i++;
            }
        }
        return n;
    }
};

File: 27-RemoveElement/27-RemoveElement.js
// Last updated: 1/13/2026, 10:49:51 PM
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let x = 0;
    for(let i = 0; i<nums.length;i++){
        if(nums[i] != val){
            nums[x] = nums[i];
            x++;
        }
    }
    return x;
};